index.js

```diff
@@ -0,0 +1,180 @@
+      i.parentElement.lastElementChild.previousElementSibling.classList.toggle('disable');
+      updateStorage();
+    });
+  });
+
+  // Add data to local storage
+  const storedData = new Task(taskValue, false, checkbox.length - 1);
+  listArray.push(storedData);
+  localStorage.setItem('list', JSON.stringify(listArray));
+
+  // Edit event listener
+  const editBtn = document.querySelectorAll('.edit-btn');
+  editBtn.forEach((i) => {
+    i.addEventListener('click', () => {
+      editTask(listItem, i.previousElementSibling);
+      // updateStorage();
+    });
+  });
+
+  //  Delete event listener
+  const removeBtn = document.querySelectorAll('.delete-btn');
+  removeBtn.forEach((i) => {
+    i.addEventListener('click', () => {
+      deleteTask(i.parentElement);
+    });
+  });
+};
+
+// add task event listener
+inputField.addEventListener('keypress', (e) => {
+  if (e.key === 'Enter' && inputField.value) {
+    e.preventDefault();
+    addTask(inputField.value);
+    inputField.value = null;
+  }
+});
+
+// edit functionality
+const editTask = (listItem, task) => {
+  const editInput = document.createElement('input');
+  const storedData = JSON.parse(localStorage.getItem('list'));
+  editInput.type = 'text';
+  editInput.className = 'editInput';
+  editInput.value = task.textContent;
+  listItem.replaceChild(editInput, task);
+  editInput.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter' && editInput.value) {
+      const result = storedData.filter((text) => text.description === task.textContent);
+      const emptyArray = [];
+      for (let i = 0; i < storedData.length; i += 1) {
+        if (storedData[i].index === result[0].index) {
+          storedData[i].description = editInput.value;
+        }
+        emptyArray.push(storedData[i]);
+        localStorage.setItem('list', JSON.stringify(emptyArray));
+      }
+      listItem.replaceChild(task, editInput);
+      task.textContent = editInput.value;
+    }
+  });
+};
+
+// Delete task from list
+const deleteTask = (task) => {
+  tasks.removeChild(task);
+  let count = -1;
+  const storedData = JSON.parse(localStorage.getItem('list'));
+  const data = Array.from(storedData).filter((i) => i.completed === false);
+  data.map((i) => i.index = count += 1);
+  localStorage.setItem('list', JSON.stringify(data));
+};
+
+// Get data from local storage
+const getData = () => {
+  data.map((i) => {
+    listArray.push(i);
+    const listItem = document.createElement('li');
+    listItem.innerHTML = `
+  <input type="checkbox" class="checkbox"/>
+  <span>${i.description}</span >
+  <i class="fa-solid fa-ellipsis-vertical edit-btn"></i>
+  <i class="fa-solid fa-trash-can delete-btn"></i>
+  `;
+    tasks.appendChild(listItem);
+
+    const editBtn = document.querySelectorAll('.edit-btn');
+    editBtn.forEach((i) => {
+      i.addEventListener('click', () => {
+        editTask(listItem, i.previousElementSibling);
+      });
```

Comment:
I suggest you use a more descriptive function parameter to make it clear what each function is doing instead of using 'i', this will improve the readability of your code